packages/react-sdk/src/components/image.tsx

```diff
@@ -1,24 +1,187 @@
 import React, { forwardRef, type ElementRef, type ComponentProps } from "react";
+import { warnOnce } from "../utils/warn-once";
 
 const defaultTag = "img";
 
-type ImageProps = ComponentProps<typeof defaultTag>;
+type ImageProps = ComponentProps<typeof defaultTag> & { quality?: number };
+
+/**
+ * max(...imageSizes) must be less then min(...deviceSizes)
+ * idea is to support images with vw size less than 100vw
+ * https://github.com/vercel/next.js/blob/canary/packages/next/shared/lib/image-config.ts
+ **/
+const imageSizes = [16, 32, 48, 64, 96, 128, 256, 384];
+const deviceSizes = [640, 750, 828, 1080, 1200, 1920, 2048, 3840];
+
+const allSizes = [...imageSizes, ...deviceSizes];
+
+/**
+ * https://developers.cloudflare.com/images/image-resizing/url-format/
+ **/
+const cloudflareImageLoader = ({
+  width,
+
+  src,
+  quality,
+}: {
+  width: number;
+  src: string;
+  quality?: number;
+}) => {
+  let resizeOrigin = "";
+
+  if (process.env.NODE_ENV !== "production") {
+    if (!allSizes.includes(width)) {
+      warnOnce("Width must be only from allowed values");
+    }
+
+    // No image transformation in development
+    resizeOrigin = "https://webstudio.is";
+  }
```

Comment:
Is this for localhost? It's strange then that `https://webstudio.is` is used, because AFAIK locally we "upload" assets to local filesystem. But not sure about that. /cc @kof

Reply:
Locally we don't have cloudflare image resize, how to test/check without?

Reply:
Locally we can emulate Cloudflare resize with imgproxy inside docker compose. And use imgproxy [loader transform]( https://github.com/webstudio-is/webstudio-designer/pull/378/files#diff-2b9aebd3d7549d8e39f1c7f9cb934b5909220081cef2f6b8cfc2fc93b712140eR21) for local development. In that case docker compose will be a requirement.

Reply:
Or stick to no transform at all just adding fake `cdn-cgi/image` route locally

Reply:
this is a good question, because if you are building locally and actually need to see the page with same sizes you would get in prod, we need to build our own resizing service that runs locally.

We will probably build it at some point in any case, but can we avoid building it now? Would loading original images be a huge problem for now? Technically we don't expect people to use webstudio locally at this stage, this is something for later, after beta rollout

Reply:
As discussed in a call:
1. lets use asset id in props/styles
2. lets build a single function that returns a path to the image depending on the environment, so we don't need a proxy locally for now
3. lets use that single function in image manager in the designer as well as in SDK components

Reply:
Having that assetId should be used at the upper level component, (not in react-sdk)
I added for now additional Image prop `optimize` defaulting to `false`

With it, image behavior is the same as it was before.

Then at the upper level, we can read the image by assetId and decide whether we need any optimizations. In case of optimization is enabled we have a default loader with a fixed image path. Or we can pass any other image `loader`.

Reply:
yes totally, react-sdk sould be on the level of a generic component library from knowledge perspective + meta data specific to webstudio